src/Features/Lsif/Generator/Generator.cs

```diff
@@ -117,11 +121,13 @@ public async Task GenerateForCompilationAsync(Compilation compilation, string pr
                     // are allowed and might flush other unrelated stuff at the same time, but there's no harm -- the "causality" ordering
                     // is preserved.
                     var documentWriter = new BatchingLsifJsonWriter(_lsifJsonWriter);
-                    var documentId = await GenerateForDocumentAsync(semanticModel, languageServices, options, topLevelSymbolsResultSetTracker, documentWriter, _idFactory);
+                    var documentId = await GenerateForDocumentAsync(document, options, topLevelSymbolsResultSetTracker, documentWriter, _idFactory);
                     topLevelSymbolsWriter.FlushToUnderlyingAndEmpty();
                     documentWriter.FlushToUnderlyingAndEmpty();
 
                     documentIds.Add(documentId);
+
+                    GC.KeepAlive(semanticModel);
```

Comment:
Should we just let GenerateForDocumentAsync ask for that and hold onto it?

Reply:
I see GenerateForDocumentAsync is already doing that too. :smile:

Reply:
yeah, as long as it's done in one place, that's likely fine.  we do want to hold onto the semantic model while we process the entire doc though so all features use the same one.

Reply:
removed.